notebook/tests/selenium/utils.py

```
@@ -382,3 +386,77 @@ def trigger_keystrokes(browser, *keys):
+            return True
+        #Only the index cell is on edit mode
+        i = 0
+        for mode in cells_mode:
+            if(i == index):
+                if(mode != 'edit'):
+                    return False
+            else:
+                if(mode == 'edit'):
+                    return False
+            i += 1
+        return True
+
+    def is_focused_on(index):
+        JS = "return $('#notebook .CodeMirror-focused textarea').length;"
+        focused_cells = notebook.browser.execute_script(JS)
+        if(index == None):
+            return focused_cells == 0
+
+        assert focused_cells == 1
+
+        JS = "return $('#notebook .CodeMirror-focused textarea')[0];"
+        focused_cell = notebook.browser.execute_script(JS)
+        JS = "return IPython.notebook.get_cell(%s).code_mirror.getInputField()"%index
+        cell = notebook.browser.execute_script(JS)
+        assert focused_cell == cell
+
+
+    #general test
+    JS = "return IPython.keyboard_manager.mode;"
+    keyboard_mode = notebook.browser.execute_script(JS)
+    JS = "return IPython.notebook.mode;"
+    notebook_mode = notebook.browser.execute_script(JS)
+
+    #validate selected cell
+    JS = "return Jupyter.notebook.get_selected_cells_indices();"
+    cell_index = notebook.browser.execute_script(JS)
+    assert cell_index == [index] #only the index cell is selected
+
+    if(mode == 'command'):
+        #validate mode
+        assert mode == keyboard_mode #keyboard mode is correct
+
+        assert is_focused_on(None)#no focused cells
+
+        assert is_only_cell_edit(None)#no cells in edit mode
+    
+    elif(mode == 'edit'):
+        #Are the notebook and keyboard manager in edit mode?
+        assert mode == keyboard_mode #keyboard mode is correct
+
+        is_focused_on(index)#The specified cell is focused
+
+        assert is_only_cell_edit(index) #The specified cell is the only one in edit mode
+    
+    else:
+        assert False #An unknown mode is send
```

If this occurs, `assert False` won't give a useful error message. It should raise an exception including the unexpected value.